apps/s_client.c

```diff
@@ -1554,6 +1558,15 @@ int s_client_main(int argc, char **argv)
         }
     }
 
+    if (localstr != NULL) {
+        if (connectstr == NULL && ! (host != NULL && port != NULL)) {
```

Comment:
This test could also be write `if (connectstr == NULL && (host == NULL || port == NULL)) {`...

Reply:
Test was broken, now removed.